### PR TITLE
more object

### PR DESCRIPTION
**Pull request by jiangzhengwenjz**

### CODE REVIEW

**Comment by jiangzhengwenjz** [comment]
I'm stopping here for lack of time.

File: src/object_start.c [hunk 1]
```diff
@@ -290,3 +372,317 @@ void ObjectDestroy(struct Task* arg0) {
...
+    s32 v4;
+
+    if (!gUnk_08351648[obj->type].unkC) return;
+    r6 = &obj->unk10;
+    if (!(obj->unkC & 0x200)) {
+        if (gUnk_02020EE0[gUnk_0203AD3C].base.unk60 == obj->unk60) {
+            if (!obj->unk10.unk0) {
+                if (obj->flags & 0x4000) {
+                    r6->unk0 = sub_0803DD58(obj->type);
+                    r6->unk8 &= 0xFFF7FFFF;
```
Comment (Cleverking2003) [inline src/object_start.c:409]: invert flags

File: src/object_start.c [hunk 2]
```diff
@@ -290,3 +372,317 @@ void ObjectDestroy(struct Task* arg0) {
...
+                    CpuCopy32(r6, &sprite, sizeof(struct Sprite));
+                    sub_0815521C(&sprite, obj->unk1);
+                } else {
+                    obj->unk10.unk0 = sub_081570B0(gUnk_08351648[obj->type].unkC);
+                    r6->unk8 &= 0xFFF7FFFF;
```
Comment (Cleverking2003) [inline src/object_start.c:414]: invert flags

File: src/object_start.c [hunk 3]
```diff
@@ -290,3 +372,317 @@ void ObjectDestroy(struct Task* arg0) {
...
+
+void sub_0809A7A4(void) {
+    s32 r2;
+    struct Object2 *objTemp, *r7 = TASK_GET_STRUCT_PTR(gCurTask, objTemp);
+    struct Sprite *r6;
+
+    if (gUnk_08351648[r7->type].unkC) {
+        r6 = &r7->unk10;
+        if (r7->flags & 1)
+            r6->unk8 &= 0xFFFFFBFF;
```
Comment (Cleverking2003) [inline src/object_start.c:454]: invert flags

File: src/object_start.c [hunk 4]
```diff
@@ -290,3 +372,317 @@ void ObjectDestroy(struct Task* arg0) {
...
+        else
+            r6->unk8 |= 0x400;
+        if (!(r7->flags & 8)) {
+            r6->unkC = gUnk_08351648[r7->type].unk14[r7->unk83].unk0;
+            r6->unk1A = gUnk_08351648[r7->type].unk14[r7->unk83].unk2;
+            if (r6->unk1B != r6->unk1A || r6->unk18 != r6->unkC) {
+                r7->unk1 = 0;
+                r7->unk2 = 0;
+                r7->flags &= 0xFFFFFFFB;
```
Comment (Cleverking2003) [inline src/object_start.c:463]: invert flags

File: src/object_start.c [hunk 5]
```diff
@@ -290,3 +372,317 @@ void ObjectDestroy(struct Task* arg0) {
...
+            }
+            r2 = sub_08155128(r6);
+            if (!r2) {
+                r7->flags |= 2;
+                if (r7->flags & 4) {
+                    r6->unk1B = 0xFF;
+                    r7->unk1 = r2;
+                    r7->unk2 = r2;
+                    r7->flags &= 0xFFFFFFFB;
```
Comment (Cleverking2003) [inline src/object_start.c:472]: invert flags

File: src/object_start.c [hunk 6]
```diff
@@ -290,3 +372,317 @@ void ObjectDestroy(struct Task* arg0) {
...
+                    sub_08155128(r6);
+                }
+            } else {
+                r7->flags &= 0xFFFFFFFD;
```
Comment (Cleverking2003) [inline src/object_start.c:476]: invert flags

**Review by Cleverking2003 — COMMENTED** [review]
It's better to write these as `value &= ~flags`

**Review by Cleverking2003 — APPROVED** [review]

**Comment by jiangzhengwenjz** [comment]
Test comment

**Comment by jiangzhengwenjz** [comment]
Test comment

**Pull request merged**